### ISSUE TITLE
multiple workers with gunicorn still synchronous

### ISSUE THREAD

**Issue by wotori** [tiangolo/fastapi#4481]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
gunicorn main:app --workers 8 --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:8000
```

### Description

Hello everyone.
According to [this](https://fastapi.tiangolo.com/deployment/server-workers/) official fastapi article I started server with this command:
```
 % gunicorn main:app --workers 8 --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:8000
```
and got the output, seems everything fine.
```
[2022-01-26 16:59:54 +0300] [105927] [INFO] Starting gunicorn 20.1.0
[2022-01-26 16:59:54 +0300] [105927] [INFO] Listening at: http://0.0.0.0:8000 (105927)
[2022-01-26 16:59:54 +0300] [105927] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2022-01-26 16:59:54 +0300] [105928] [INFO] Booting worker with pid: 105928
[2022-01-26 16:59:54 +0300] [105929] [INFO] Booting worker with pid: 105929
[2022-01-26 16:59:54 +0300] [105930] [INFO] Booting worker with pid: 105930
[2022-01-26 16:59:54 +0300] [105931] [INFO] Booting worker with pid: 105931
[2022-01-26 16:59:54 +0300] [105932] [INFO] Booting worker with pid: 105932
[2022-01-26 16:59:54 +0300] [105933] [INFO] Booting worker with pid: 105933
[2022-01-26 16:59:55 +0300] [105934] [INFO] Booting worker with pid: 105934
[2022-01-26 16:59:55 +0300] [105935] [INFO] Booting worker with pid: 105935
[2022-01-26 16:59:55 +0300] [105928] [INFO] Started server process [105928]
[2022-01-26 16:59:55 +0300] [105928] [INFO] Waiting for application startup.
[2022-01-26 16:59:55 +0300] [105928] [INFO] Application startup complete.
[2022-01-26 16:59:56 +0300] [105931] [INFO] Started server process [105931]
[2022-01-26 16:59:56 +0300] [105931] [INFO] Waiting for application startup.
[2022-01-26 16:59:56 +0300] [105931] [INFO] Application startup complete.
...
```

then I opened 4 terminal windows and asking the same request each window. The response appear one by one in each terminal window and the same thing happen even if I use two computers for test, so It's not work as expected (in parallel). What am I doing wrong?
![image](https://user-images.githubusercontent.com/10486621/151178546-e303ba2a-5958-46e1-9017-9f1c6daaf715.png)

### Operating System

Linux

### FastAPI Version

0.63.0

### Python Version

3.9.10

### Additional Context
I have 4 cpu and 8 threads
![image](https://user-images.githubusercontent.com/10486621/151180880-4c4bc8a2-bfec-45b4-a1f7-8506ff0c229c.png)

**Comment by STeveShary** [comment]
@wotori Based on the information provided, it's difficult for us to understand what is the specific problem.  I agree that @jgould22 is probably asking the correct question:

> Are you blocking the event loop ?

Could you answer the following questions:

1.  How are to talking to a database?  Are you using `async/await` on this?
2. How are you talking to an external service? Are you using `async/await` on this?
3. Do your route functions have a `async` prefix?

Moving to async can be confusing to start using.  I have teams that accidentally write blocking code and we just have to make small adjustments to make them async.

**Comment by STeveShary** [comment]
> I thought when I use workers it's a completely separate process and they should work in parallel and gunicorn just manage requests and point them to free instances, Isn't It correct?
Yes, this is correct.  According to your setup, you should be able to handle 8 calls in parallel, _even_ if your endpoints are blocking.
> What the purpose of workers If they don't work without async/await?
They do work without async/await.


Try this code below with your gunicorn/uvicorn setup.  It should allow you to make one call per second on the block endpoint.  Both should have latency at roughly 1,000 milliseconds.

```python3
import asyncio
import time

from fastapi import FastAPI

app = FastAPI()


@app.get("/blocking")
def blocking_route():
    time.sleep(1)


@app.get("/nonBlocking")
async def blocking_route():
    await asyncio.sleep(1)

```

You should be able to run apache benchmark to get latency of around 1,000ms:

```bash
ab -n 80 -c 8 http://localhost:8000/blocking
```

**Comment by raphaelauv** [comment]
if you have a question or issue with gunicorn -> https://github.com/benoitc/gunicorn

or uvicorn -> https://github.com/encode/uvicorn

it does not look like a fastapi issue

**Comment by wotori** [comment]
> > I thought when I use workers it's a completely separate process and they should work in parallel and gunicorn just manage requests and point them to free instances, Isn't It correct?
> > Yes, this is correct.  According to your setup, you should be able to handle 8 calls in parallel, _even_ if your endpoints are blocking.
> > What the purpose of workers If they don't work without async/await?
> > They do work without async/await.
> 
> Try this code below with your gunicorn/uvicorn setup. It should allow you to make one call per second on the block endpoint. Both should have latency at roughly 1,000 milliseconds.
> 
> ```python
> import asyncio
> import time
> 
> from fastapi import FastAPI
> 
> app = FastAPI()
> 
> 
> @app.get("/blocking")
> def blocking_route():
>     time.sleep(1)
> 
> 
> @app.get("/nonBlocking")
> async def blocking_route():
>     await asyncio.sleep(1)
> ```
> 
> You should be able to run apache benchmark to get latency of around 1,000ms:
> 
> ```shell
> ab -n 80 -c 8 http://localhost:8000/blocking
> ```

Got your point, thanks for share the Idea! 
I did asyncio In a wrong way, now everything works.
